libs/blocks/carousel/carousel.js

```diff
@@ -378,6 +378,16 @@ export default function init(el) {
 
   el.append(...nextPreviousBtns, controlsContainer);
 
+  function handleImageLoad() {
+    if (!controlsContainer.classList.contains('d-none') && nextPreviousBtns.findIndex((btnEl) => btnEl.classList.contains('d-none')) === -1) this.removeEventListener('load', handleImageLoad, false);
+    else {
+      if (controlsContainer.classList.contains('d-none')) controlsContainer.classList.remove('d-none');
+      nextPreviousBtns.forEach((btnEl) => {
+        if (btnEl.classList.contains('d-none')) btnEl.classList.remove('d-none');
+      });
+    }
+  }
```

Comment:
I'm not particularly happy with this solution, it adds a lot of code paths and makes maintainability worse.